### PR TITLE
Improved mp3 player

### PR DESCRIPTION
**Pull request by KrauseFx**
- Proper escaping of paths
- New directory parameter
- Parameter input validation
- Added screenshots

### CODE REVIEW

File: lib/fastlane/plugin/tunes/actions/tunes_action.rb [hunk 1]
```diff
@@ -1,25 +1,56 @@
...
 module Fastlane
   module Actions
     class TunesAction < Action
       def self.run(params)
-        sh "afplay '#{params[:file_path]}'"
+        if params[:file_path]
+          play_song(params[:file_path])
+        elsif params[:directory]
+          UI.message("Playing all songs from directory '#{params[:directory]}'")
+          Dir[File.join(params[:directory], "*.mp3")].each do |current|
```
Comment (neonichu) [inline lib/fastlane/plugin/tunes/actions/tunes_action.rb:10]: We should have a bunch more file extensions here, afplay manpage should list all the supported ones

File: lib/fastlane/plugin/tunes/actions/tunes_action.rb [hunk 2]
```diff
@@ -1,25 +1,56 @@
...
+      end
+
+      def self.play_song(path)
+        Helper::TunesHelper.print_audio_information(path)
+        sh "afplay #{path.shellescape}"
       end
 
       def self.description
-        "Play music using fastlane, because you can."
+        "Play music using fastlane, because you can"
```
Comment (neonichu) [inline lib/fastlane/plugin/tunes/actions/tunes_action.rb:24]: Leave the dot alone

File: lib/fastlane/plugin/tunes/actions/tunes_action.rb [hunk 3]
```diff
@@ -1,25 +1,56 @@
...
       end
 
       def self.authors
-        ["Boris Bügling"]
+        ["neonacho", "krausefx"]
```
Comment (neonichu) [inline lib/fastlane/plugin/tunes/actions/tunes_action.rb:28]: :-1:

File: lib/fastlane/plugin/tunes/actions/tunes_action.rb (hunk 2 above)
Comment (KrauseFx) [inline lib/fastlane/plugin/tunes/actions/tunes_action.rb:24]: Nope, no dots allowed: https://github.com/fastlane/fastlane/blob/master/fastlane/spec/actions_list_spec.rb#L33
I shall add a linter for that

File: lib/fastlane/plugin/tunes/actions/tunes_action.rb (hunk 1 above)
Comment (KrauseFx) [inline lib/fastlane/plugin/tunes/actions/tunes_action.rb:10]: No!

File: lib/fastlane/plugin/tunes/actions/tunes_action.rb (hunk 3 above)
Comment (KrauseFx) [inline lib/fastlane/plugin/tunes/actions/tunes_action.rb:28]: Best!

**Comment by KrauseFx** [comment]
Pls merge

File: lib/fastlane/plugin/tunes/actions/tunes_action.rb (hunk 2 above)
Comment (neonichu) [inline lib/fastlane/plugin/tunes/actions/tunes_action.rb:24]: :sob:

**Comment by neonichu** [comment]
Pls fix

**Comment by KrauseFx** [comment]
It's ready, please merge 👍

File: lib/fastlane/plugin/tunes/actions/tunes_action.rb (hunk 1 above)
Comment (neonichu) [inline lib/fastlane/plugin/tunes/actions/tunes_action.rb:10]: Yes.

**Comment by KrauseFx** [comment]
Okay, updated, please merge 🚀

**Comment by neonichu** [comment]
:trollface:

**Comment by KrauseFx** [comment]
Not merged yet 😢

**Comment by neonichu** [comment]
Improved and stolen as 80c1f293af58aaa908251dc4662a71cf9e5d8730

**Comment by KrauseFx** [comment]
Wow

**Pull request closed without merging**